lib/array.js

```diff
@@ -38,6 +38,59 @@ const map = (
   }
 };
 
+const NS_PER_SEC = 1e9;
+const DEFAULT_OPTIONS = { min: 5, percent: 0.7 };
+
+const asyncMap = (
+  // Non-blocking map
+  items, // array, incoming
+  fn, // function, (value, index) => {}
+  options, // object (optional), { min, percent }
+  done // function (optional), on done callback function(result)
+) => {
+  if (typeof options === 'function') done = options;
+  const { min, percent } = Object.assign(DEFAULT_OPTIONS, options);
```

Comment:
This will change `DEFAULT_OPTIONS`, you should create (assign to) a new object `{}` or destructure them manually.